vulkan_helpers/vulkan_application.h

```diff
@@ -115,13 +116,35 @@ struct VulkanApplicationOptions {
     return *this;
   }
   VulkanApplicationOptions& EnableVulkan11() {
-    use_vulkan_1_1 = true;
+    this->SetVulkanApiVersion(VK_API_VERSION_1_1);
     return *this;
   }
   VulkanApplicationOptions& Enable10BitHDR() {
     use_10bit_hdr = true;
     return *this;
   }
+  
+  VulkanApplicationOptions& SetVulkanApiVersion(uint32_t vulkan_api_version) {
+    uint32_t variant = VK_API_VERSION_VARIANT(vulkan_api_version);
+    uint32_t major = VK_API_VERSION_MAJOR(vulkan_api_version);
+    uint32_t minor = VK_API_VERSION_MINOR(vulkan_api_version);
+    uint32_t patch = VK_API_VERSION_PATCH(vulkan_api_version);
+
+    static_assert(
+        VK_API_VERSION_MAJOR(VK_HEADER_VERSION_COMPLETE) == 1,
+        "review the following to make sure that the version check is correct "
+        " after updating vulkan.h");
+
+    if (variant != 0 ||
+        major > VK_API_VERSION_MAJOR(VK_HEADER_VERSION_COMPLETE) ||
+        minor > VK_API_VERSION_MINOR(VK_HEADER_VERSION_COMPLETE) ||
+        vulkan_api_version > VK_HEADER_VERSION_COMPLETE) {
```

Comment:
Can we get of the `major` and `minor` checks? It seems like we just want to check that variant is 0 and the requested Vulkan version is not newer than `VK_HEADER_VERSION_COMPLETE`.

Reply:
Or probably I can get rid of all the checks and leave it to validation layer.

Reply:
I would get rid of most of these checks. I think it can be left to vkCreateInstance and/or the validation layer to provide the proper feedback if incorrect or unsupported values are used.

Reply:
Those are now removed